azure_arc_k8s_jumpstart/cluster_api/capi_keyvault_extension/capi_keyvault_k8s_extension.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
+
+# <--- Change the following environment variables according to your Azure service principal name --->
+export appId='<Your Azure service principal name>'
+export password='<Your Azure service principal password>'
+export tenantId='<Your Azure tenant ID>'
+export keyVaultResourceGroup='<KeyVault Resource Group name>'
+export keyVaultLocation='<Key Vault Location>'
+export resourceGroup='<Azure Arc CLuster Resource Group name>'
+export arcClusterName='<Azure Arc Cluster Name>'
+export namespace='hello-arc'
+export k8sExtensionName='sscsi'
+export keyVaultName=secret-store-$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 10 | head -n 1)
+
+# Installing Helm 3
+echo "Installing Helm 3"
+curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3
+chmod 700 get_helm.sh
+./get_helm.sh
+
+# Installing Azure CLI & Azure Arc Extensions
+echo "Installing Azure CLI"
+curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
+
+echo "Clear cached helm Azure Arc Helm Charts"
+rm -rf ~/.azure/AzureArcCharts
+
+echo "Login to Az CLI using the service principal"
+az login --service-principal --username $appId --password $password --tenant $tenantId
+
+echo "Checking if you have up-to-date Azure Arc Az CLI 'connectedk8s' extension..."
+az extension show --name "connectedk8s" &> extension_output
+if cat extension_output | grep -q "not installed"; then
+az extension add --name "connectedk8s"
+rm extension_output
+else
+az extension update --name "connectedk8s"
+rm extension_output
+fi
+echo ""
+
+echo "Checking if you have up-to-date Azure Arc Az CLI 'k8s-extension' extension..."
+az extension show --name "k8s-extension" &> extension_output
+if cat extension_output | grep -q "not installed"; then
+az extension add --name "k8s-extension"
+rm extension_output
+else
+az extension update --name "k8s-extension"
+rm extension_output
+fi
+echo ""
+
+echo "Creating Key Vault"
+az group create --name $keyVaultResourceGroup --location $keyVaultLocation
+az keyvault create --name $keyVaultName --resource-group $keyVaultResourceGroup --location $keyVaultLocation
+
+echo "Creating Key Vault Secret"
+az keyvault secret set --vault-name $keyVaultName --name dbusername --value "HelloArc!"
```

Comment:
You would need to use single quotes for the value `'HelloArc!'` or escape the `!` in double quotes `"HelloArc\!"`

Reply:
Updated to single Quote